numpy/ma/core.py

```diff
@@ -3305,7 +3305,8 @@ def _scalar_heuristic(arr, elem):
                     # which should have shape () to avoid breaking several
                     # methods. There is no great way out, so set to
                     # first element.  See issue #6723.
-                    if dout._fill_value.ndim > 0:
+                    dtype = self._fill_value.dtype[indx]
+                    if dtype.ndim > 0:
```

Comment:
Is this change necessary?

I'm tempted to add `assert isinstance(self._fill_value, np.ndarray) before this line.

Reply:
Sorry I'm kinda confused here, so if the _fill_value is a string instead of ndarray, are we suppose to throw an error or change it to a 0darray?

Reply:
`_fill_value` should never be anything other than a 0d array. If it's somehow a string here, then something else has a bug, and we should track down that bug.

Reply:
Ah I see, guess I was not understanding the expected behavior correctly.